CI: Runtime Test (privileged) - Test_LocalNodeWatcher/update

## CI failure

Hit on https://github.com/cilium/cilium/pull/34091
Link: https://github.com/cilium/cilium/actions/runs/10163720493/job/28107529401

```
--- FAIL: Test_LocalNodeWatcher (0.00s)
    --- FAIL: Test_LocalNodeWatcher/update (0.00s)
        local_node_watcher_test.go:79: 
            	Error Trace:	/host/pkg/hubble/observer/local_node_watcher_test.go:79
            	Error:      	Not equal: 
            	            	expected: []string{"kubernetes.io/arch=arm64", "kubernetes.io/hostname=kind-kind", "kubernetes.io/os=linux"}
            	            	actual  : []string{"kubernetes.io/arch=amd64", "kubernetes.io/hostname=ip-1-2-3-4.us-west-2.compute.internal", "kubernetes.io/os=linux", "topology.kubernetes.io/region=us-west-2", "topology.kubernetes.io/zone=us-west-2d"}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,5 +1,7 @@
            	            	-([]string) (len=3) {
            	            	- (string) (len=24) "kubernetes.io/arch=arm64",
            	            	- (string) (len=32) "kubernetes.io/hostname=kind-kind",
            	            	- (string) (len=22) "kubernetes.io/os=linux"
            	            	+([]string) (len=5) {
            	            	+ (string) (len=24) "kubernetes.io/arch=amd64",
            	            	+ (string) (len=60) "kubernetes.io/hostname=ip-1-2-3-4.us-west-2.compute.internal",
            	            	+ (string) (len=22) "kubernetes.io/os=linux",
            	            	+ (string) (len=39) "topology.kubernetes.io/region=us-west-2",
            	            	+ (string) (len=38) "topology.kubernetes.io/zone=us-west-2d"
            	            	 }
            	Test:       	Test_LocalNodeWatcher/update
FAIL
```

--- a/pkg/hubble/observer/local_node_watcher_test.go
+++ b/pkg/hubble/observer/local_node_watcher_test.go
@@ -54,10 +54,11 @@ func Test_LocalNodeWatcher(t *testing.T) {
 	}
 
 	var watcher *LocalNodeWatcher
+	store := node.NewTestLocalNodeStore(localNode)
 
 	t.Run("NewLocalNodeWatcher", func(t *testing.T) {
 		var err error
-		watcher, err = NewLocalNodeWatcher(ctx, node.NewTestLocalNodeStore(localNode))
+		watcher, err = NewLocalNodeWatcher(ctx, store)
 		require.NoError(t, err)
 		require.NotNil(t, watcher)
 	})
@@ -71,7 +72,9 @@ func Test_LocalNodeWatcher(t *testing.T) {
 	})
 
 	t.Run("update", func(t *testing.T) {
-		watcher.update(updatedNode)
+		store.Update(func(ln *node.LocalNode) {
+			*ln = updatedNode
+		})
 		var flow flowpb.Flow
 		stop, err := watcher.OnDecodedFlow(ctx, &flow)
 		require.False(t, stop)

